Update the metric `Export` interface to accept a `*ResourceMetrics` instead of `ResourceMetrics` (#3853)
open-telemetry/opentelemetry-go: CHANGELOG.md

- Avoid creating new objects on all calls to `WithDeferredSetup` and `SkipContextSetup` in OpenTracing bridge. (#3833)
- The `New` and `Detect` functions from `go.opentelemetry.io/otel/sdk/resource` return errors that wrap underlying errors instead of just containing the underlying error strings. (#3844)

> - The metric `Export` interface to accepts a `*ResourceMetrics` instead of `ResourceMetrics` (#3853)

Reviewer: ```suggestion
- The `Export` interface in `go.opentelemetry.io/otel/exporters/stdout/stdoutmetric` to accept `*metricdata.ResourceMetrics` instead of `metricdata.ResourceMetrics`. (#3853)
```

I think this makes it easier to understand where the change happened.

Reviewer 2: The interface is in `go.opentelemetry.io/otel/sdk/metric`

Reviewer: You're right, thanks!

> - The metric `Export` interface to accepts a `*ResourceMetrics` instead of `ResourceMetrics` (#3853)

Reviewer 3: Can you mention the package (full path) like in other changelog entries?

Author: updated as suggested

- Avoid creating new objects on all calls to `WithDeferredSetup` and `SkipContextSetup` in OpenTracing bridge. (#3833)
- The `New` and `Detect` functions from `go.opentelemetry.io/otel/sdk/resource` return errors that wrap underlying errors instead of just containing the underlying error strings. (#3844)
- Both the `Histogram` and `HistogramDataPoint` are redefined with a generic argument of `[N int64 | float64]` in `go.opentelemetry.io/otel/sdk/metric/metricdata`. (#3849)

> - The metric `Export` interface from `go.opentelemetry.io/otel/sdk/metric` accepts a `*ResourceMetrics` instead of `ResourceMetrics` (#3853)
> 

Reviewer 2: ```suggestion
- The metric `Export` interface from `go.opentelemetry.io/otel/sdk/metric` accepts a `*ResourceMetrics` instead of `ResourceMetrics`. (#3853)
```
